### DISCUSSION TITLE
[Suggestion] Support Android container

### DISCUSSION THREAD

**Discussion by momobobe** [89luca89/distrobox#1893] [Ideas]
**Is your feature request related to a problem? Please describe.**
[WayDroid](https://waydro.id/) is a container solution for running Android App on Linux.

**Describe the solution you'd like**
I hope Distrobox can support it as a container distro so Apps installed there can be managed in a same place.

**Describe alternatives you've considered**
~~No idea~~ Update: just found another Android container called [ReDroid](https://github.com/remote-android/redroid-doc) which runs with docker or podman

**Additional context**
WayDroid works with lxc instead of Docker or Podman, not sure, but maybe it's possible to find a way to support it.

**Comment by 89luca89** [comment]
Not familiar with it, but if it doesn't run with docker/podman I don't think there will be any way to support it in the short run

**Comment by momobobe** [comment]
@89luca89 I see. Just found another project [ReDroid](https://github.com/remote-android/redroid-doc) which runs with docker or podman. Maybe that's feasible. Anyway I think an Android container could be quite useful, considering how many apps are there

**Comment by momobobe** [comment]
Just found duplicated with #741

**Comment by 89luca89** [comment]
Redroid seems nice, but needs kernel modules and a lot of workarounds
So I cannot add it to the official compatibility list

But theoretically one could already try to use it

**Comment by profetik-777** [comment]
@89luca89 Maybe we should create a new tag or place for future enhancements that you know for sure you WANT to do, but are pending any real time lines / release roadmap. Maybe "Someday" tag?

**Comment by 89luca89** [comment]
Done :joy:

**Comment by scottsweb** [comment]
I haven't got around to testing this yet, but the [vanilla-os folks are working on a waydroid container](https://github.com/Vanilla-OS/waydroid-image). I have scaffolded a `distrobox.ini` with the following contents:

```
[waydroid]
image=ghcr.io/vanilla-os/waydroid:main
entry=false
pull=true
root=true
init=true
replace=false
start_now=true
unshare_all=true
# exports
exported_bins="/usr/bin/ewaydroid"
exported_bins_path="~/.local/bin"
```

and they provide this example in their docs:

```
distrobox create --root --init --unshare-all -i ghcr.io/vanilla-os/waydroid:main -n waydroid
distrobox enter --root waydroid
```

I will report back once I have found time to experiment. It would also be great to have a new version of distrobox available so the `exported_bins` work as expected.

**Comment by Nitrousoxide** [comment]
Does this save the basic initialization and the one for gapps or in a persistent volume claim?  We wouldn't want to have to do that every time the container is redeployed.

**Comment by Pacific6938** [comment]
tried running waydroid using your guide but cant get it working also doesent work when installing it in archlinux or fedora
https://distrobox.it/useful_tips/#automated-installation


[user@waydroid ~]$ ewaydroid show-full-ui
Session:        STOPPED
[18:08:45] RuntimeError: Command failed: % mount -o ro /var/lib/waydroid/images/system.img /var/lib/waydroid/rootfs
^C
[user@waydroid ~]$ sudo ewaydroid show-full-ui
touch: cannot touch '/root/.config/waydroid-props': No such file or directory
/usr/bin/ewaydroid: line 8: /root/.config/waydroid-props: No such file or directory
Session:        STOPPED
[18:08:53] ERROR: org.freedesktop.DBus.Error.NoServer: Failed to connect to socket /run/host/: Connection refused
[18:08:53] See also: <https://github.com/waydroid>
Run 'waydroid log' for details.
[18:09:00] WayDroid session is stopped
[18:09:00] WayDroid session is stopped
[18:09:00] WayDroid session is stopped
[18:09:00] WayDroid session is stopped
[18:09:00] WayDroid session is stopped
[18:09:00] WayDroid session is stopped
[18:09:00] Starting waydroid session
[18:09:00] ERROR: org.freedesktop.DBus.Error.NoServer: Failed to connect to socket /run/host/: Connection refused
[18:09:00] See also: <https://github.com/waydroid>
Run 'waydroid log' for details.

**Comment by dottorblaster** [comment]
Moving this to discussions. Feel free to keep discussing the topic!
